File: server/demos/php/farmacia/app/Models/DetalleCompra.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class DetalleCompra extends Model
{
    protected $table = 'detalle_compra';

    protected $fillable = [
        'compra_id', 'producto_id',
        'numero_lote', 'fecha_vencimiento',
        'cantidad', 'precio_unitario', 'subtotal',
    ];

    public function compra(): BelongsTo
    {
        return $this->belongsTo(Compra::class);
    }

    public function producto(): BelongsTo
    {
        return $this->belongsTo(Producto::class);
    }
}

File: server/demos/dotnet/minimarket/MiniMarket.Web/Models/DetalleCompra.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MiniMarket.Web.Models
{
    public class DetalleCompra
    {
        [Key]
        public int Id { get; set; }

        public int CompraId { get; set; }
        public virtual Compra Compra { get; set; }

        public int ProductoId { get; set; }
        public virtual Producto Producto { get; set; }

        public int Cantidad { get; set; }

        // --- NORMALIZAMOS EL NOMBRE A 'PrecioUnitario' ---
        [Column(TypeName = "decimal(18,2)")]
        public decimal PrecioUnitario { get; set; } 

        [Column(TypeName = "decimal(18,2)")]
        public decimal Total { get; set; }
    }
}